### PR TITLE
Institution rating calculation improvements

### PR DESCRIPTION
**Pull request by devin-mccurdy**
## Description
"NA" votes are no longer included in vote counts. This change meant making sure that average ratings for institutions with only "NA" votes are defaulted to `null` to prevent the average calculation from dividing by zero.

[ZenHub Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/13635)

## Testing done
Tested locally and QA reviewed

## Screenshots
N/A

## Acceptance criteria
- [x] Ratings are calculated correctly

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: app/models/institution_builder.rb [hunk 1]
```diff
@@ -800,11 +807,15 @@ def self.build_institution_category_ratings_for_category(version_id, category)
         SUM(CASE #{category} WHEN 4 THEN 1 ELSE 0 END),
         SUM(CASE #{category} WHEN 5 THEN 1 ELSE 0 END),
         SUM(CASE WHEN #{category} IS NULL THEN 1 ELSE 0 END),
+        CASE
+          WHEN COUNT(#{category}) = 0 THEN NULL
```
Comment (thilton-oddball) [inline app/models/institution_builder.rb:811]: I'm rusty when it comes to case statements but should this be a double equal sign?

WHEN COUNT(#{category}) == 0 THEN NULL?

File: app/models/institution_builder.rb [hunk 2]
```diff
@@ -841,10 +856,15 @@ def self.build_institution_category_ratings(version_id)
       FROM(
         SELECT
           institution_id,
+          CASE
+            WHEN SUM(rated5_count) + SUM(rated4_count) + SUM(rated3_count)
+              + SUM(rated2_count) + SUM(rated1_count) = 0 THEN NULL::float
```
Comment (thilton-oddball) [inline app/models/institution_builder.rb:861]: Should == be used here also? Again, I'm rusty on case statements, so I could be wrong.

**Review by thilton-oddball — APPROVED** [review]
Have a couple of comments that should be double checked but otherwise looks good.

**Pull request merged**